src/methods/geom_relations/disjoint.jl

```diff
@@ -1,42 +1,247 @@
+implementation based on the geometry trait.
+
+Each of these calls a method in the geom_geom_processors file. The methods in
+this file determine if the given geometries meet a set of criteria. For the
+`disjoint` function and arguments g1 and g2, this criteria is as follows:
+    - points of g1 are not allowed to be in the interior of g2
+    - points of g1 are not allowed to be on the boundary of g2
+    - points of g1 are allowed to be in the exterior of g2
+    - no points required to be in the interior of g2
+    - no points of g1 are required to be on the boundary of g2
+    - no points of g1 are required to be in the exterior of g2
+
+The code for the specific implementations is in the geom_geom_processors file.
+=#
+
+const DISJOINT_ALLOWS = (in_allow = false, on_allow = false, out_allow = true)
+const DISJOINT_CURVE_ALLOWS = (over_allow = false, cross_allow = false, on_allow = false, out_allow = true)
+const DISJOINT_REQUIRES = (in_require = false, on_require = false, out_require = false)
 """
     disjoint(geom1, geom2)::Bool
 
-Return `true` if the intersection of the two geometries is an empty set.
+Return `true` if the first geometry is disjoint from the second geometry.
 
-# Examples
+Return `true` if the first geometry is disjoint from the second geometry. The
+interiors and boundaries of both geometries must not intersect.
 
-```jldoctest
+## Examples
+```jldoctest setup=:(using GeometryOps, GeometryBasics)
 import GeometryOps as GO, GeoInterface as GI
 
-poly = GI.Polygon([[(-1, 2), (3, 2), (3, 3), (-1, 3), (-1, 2)]])
-point = (1, 1)
-GO.disjoint(poly, point)
+line = GI.LineString([(1, 1), (1, 2), (1, 3), (1, 4)])
+point = (2, 2)
+GO.disjoint(point, line)
 
 # output
 true
 ```
 """
-disjoint(g1, g2)::Bool = disjoint(trait(g1), g1, trait(g2), g2)
+disjoint(g1, g2)::Bool = _disjoint(trait(g1), g1, trait(g2), g2)
+
+# # Convert features to geometries
 disjoint(::FeatureTrait, g1, ::Any, g2)::Bool = disjoint(GI.geometry(g1), g2)
 disjoint(::Any, g1, t2::FeatureTrait, g2)::Bool = disjoint(g1, geometry(g2))
-disjoint(::PointTrait, g1, ::PointTrait, g2)::Bool = !point_equals_point(g1, g2)
-disjoint(::PointTrait, g1, ::LineStringTrait, g2)::Bool = !point_on_line(g1, g2)
-disjoint(::PointTrait, g1, ::PolygonTrait, g2)::Bool = !point_in_polygon(g1, g2)
-disjoint(::LineStringTrait, g1, ::PointTrait, g2)::Bool = !point_on_line(g2, g1)
-disjoint(::LineStringTrait, g1, ::LineStringTrait, g2)::Bool = !line_on_line(g1, g2)
-disjoint(::LineStringTrait, g1, ::PolygonTrait, g2)::Bool = !line_in_polygon(g2, g1)
-disjoint(::PolygonTrait, g1, ::PointTrait, g2)::Bool = !point_in_polygon(g2, g1)
-disjoint(::PolygonTrait, g1, ::LineStringTrait, g2)::Bool = !line_in_polygon(g2, g1)
-disjoint(::PolygonTrait, g1, ::PolygonTrait, g2)::Bool = polygon_disjoint(g2, g1)
-
-function polygon_disjoint(poly1, poly2)
-    for point in GI.getpoint(poly1)
-        point_in_polygon(point, poly2) && return false
+
```

Comment:
I think we need this for ambiguity:
```suggestion
disjoint(::FeatureTrait, g1, t2::FeatureTrait, g2)::Bool = disjoint(GI.geometry(g1), GI.geometry(g2))
```